iceoryx_posh/test/moduletests/test_popo_active_call_set.cpp

```diff
@@ -0,0 +1,979 @@
+    ASSERT_TRUE(result.has_error());
+    EXPECT_THAT(result.get_error(), Eq(ActiveCallSetError::EVENT_ALREADY_ATTACHED));
+}
+
+TEST_F(ActiveCallSet_test, OverridingAlreadyAttachedEventWithoutEnumFails)
+{
+    m_sut->attachEvent(m_simpleEvents[0U], ActiveCallSet_test::triggerCallback<0U>);
+
+    auto result = m_sut->attachEvent(m_simpleEvents[0U], ActiveCallSet_test::triggerCallback<0U>);
+    ASSERT_TRUE(result.has_error());
+    EXPECT_THAT(result.get_error(), Eq(ActiveCallSetError::EVENT_ALREADY_ATTACHED));
+}
+
+TEST_F(ActiveCallSet_test, AttachingSameClassWithTwoDifferentEventsWorks)
+{
+    m_sut->attachEvent(m_simpleEvents[0U], SimpleEvent::Hypnotoad, ActiveCallSet_test::triggerCallback<0U>);
+
+    EXPECT_FALSE(m_sut
+                     ->attachEvent(m_simpleEvents[0U],
+                                   SimpleEvent::StoepselBachelorParty,
+                                   ActiveCallSet_test::triggerCallback<0U>)
+                     .has_error());
+}
+
+TEST_F(ActiveCallSet_test, DetachingSameClassWithDifferentEventEnumChangesNothing)
+{
+    m_sut->attachEvent(m_simpleEvents[0U], SimpleEvent::Hypnotoad, ActiveCallSet_test::triggerCallback<0U>);
+
+    m_sut->detachEvent(m_simpleEvents[0U], SimpleEvent::StoepselBachelorParty);
+    EXPECT_THAT(m_sut->size(), Eq(1U));
+}
+
+TEST_F(ActiveCallSet_test, DetachingDifferentClassWithSameEventEnumChangesNothing)
+{
+    m_sut->attachEvent(m_simpleEvents[0U], SimpleEvent::Hypnotoad, ActiveCallSet_test::triggerCallback<0U>);
+
+    m_sut->detachEvent(m_simpleEvents[1U], SimpleEvent::Hypnotoad);
+    EXPECT_THAT(m_sut->size(), Eq(1U));
+}
+
+TEST_F(ActiveCallSet_test, AttachingWithoutEnumTillCapacityFilledSetsUpNoEventEnumTriggerHandle)
+{
+    fillUpWithSimpleEvents();
+
+    for (uint64_t i = 0U; i < m_sut->capacity(); ++i)
+    {
+        EXPECT_TRUE(m_simpleEvents[i].m_handleNoEventEnum.isValid());
+    }
+}
+
+TEST_F(ActiveCallSet_test, DTorDetachesAllAttachedEvents)
+{
+    fillUpWithSimpleEvents();
+
+    auto capacity = m_sut->capacity();
+    m_sut.reset();
+
+    for (uint64_t i = 0U; i < capacity; ++i)
+    {
+        EXPECT_FALSE(m_simpleEvents[i].m_handleHypnotoad.isValid());
+    }
+}
+
+TEST_F(ActiveCallSet_test, AttachedEventDTorDetachesItself)
+{
+    {
+        SimpleEventClass fuu;
+        m_sut->attachEvent(fuu, ActiveCallSet_test::triggerCallback<0U>);
+    }
+
+    EXPECT_THAT(m_sut->size(), Eq(0U));
+}
+
+TEST_F(ActiveCallSet_test, AttachingSimpleEventWithoutEnumSetsNoEventEnumTriggerHandle)
+{
+    SimpleEventClass fuu;
+    m_sut->attachEvent(fuu, ActiveCallSet_test::triggerCallback<0U>);
+
+    EXPECT_TRUE(static_cast<bool>(fuu.m_handleNoEventEnum));
+}
+
+TEST_F(ActiveCallSet_test, DetachingSimpleEventResetsTriggerHandle)
+{
+    SimpleEventClass fuu;
+    m_sut->attachEvent(fuu, ActiveCallSet_test::triggerCallback<0U>);
+    m_sut->detachEvent(fuu);
+
+    EXPECT_FALSE(static_cast<bool>(fuu.m_handleHypnotoad));
```

Comment:
Why do you use the static_cast instead of ``isValid()``?
Shouldn't it be ``m_handleNoEventEnum``?